File: golang/twoSum.go
package main

import "fmt"

func twoSum(nums [4]int, target int) []int {
    lookup := make(map[int]int)
    for i, v := range nums {
        j, ok := lookup[-v]
        lookup[v - target] = i
        if ok {
            return []int{j, i}
        }
    }
    return []int{}
}


func main(){
	nums := [...]int{2, 7, 11, 15}
	target := 9
	x := twoSum(nums, target)
	fmt.Println(x)
}

File: python/twoSum.py
def twoSum(nums, target):
	mapper = {}
	for i, c in enumerate (nums):
		rest = target - c
		if rest in mapper:
			return [i, mapper[rest]]
		else:
			mapper[c] = i 
twoSum([2,7,11,15], 9)
